stack/stack.go

```diff
@@ -43,11 +50,21 @@ func ParseYAML(yamlFile string) (*Services, error) {
 		return nil, fmt.Errorf("'%s' is the only valid provider for this tool - found: %s", providerName, services.Provider.Name)
 	}
 
+	if len(filter) > 0 {
+		for k, function := range services.Functions {
+			function.Name = k
+			match, _ := regexp.MatchString(filter, function.Name)
+			if !match {
+				delete(services.Functions, function.Name)
+			}
+		}
+	}
+
 	return &services, err
```

Comment:
Shouldn't the error be `nil` here?

Reply:
Do you mean I should set the `err` to `nil` explicitly before returning it?

Reply:
for clarity yes